File: AccTelemetryTracker.Datastore/Models/MotecFile.cs
namespace AccTelemetryTracker.Datastore.Models;

public class MotecFile
{
    public int Id { get; set; }
    
    public int CarId { get; set; }

    public Car Car { get; set; }
    
    public int TrackId { get; set; }

    public Track Track { get; set; }
    
    public DateTime DateInserted { get; set; }

    public DateTime SessionDate { get; set; }
    
    public int NumberOfLaps { get; set; }
    
    public double FastestLap { get; set; }
    
    public string FileLocation { get; set; } = string.Empty;
}

File: client/acc-telemetry-tracker/src/app/_models/motecFile.ts
export interface MotecFile {
    id: number;
    carName: string;
    carClass: string;
    trackName: string;
    username: string;
    dateInserted: Date;
    sessionDate: Date;
    numberOfLaps: number;
    fastestLap: number;
    comment: string;
    changedComment: string;
    editingComment: boolean;
    editingConditions: boolean;
    trackConditions: string;
    gameVersion: string;
    changedTrackConditions: string;
    laps: MotecLap[];
}

export interface MotecLap {
    lapNumber: number;
    lapTime: number;
    sessionTime: number;
}

export interface MotecLaps {
    carTrackAverageLap: number;
    classAverageLap: number;
    classBestLap: number;
    laps: MotecLap[];
}

export interface MotecStat {
    car: string;
    carId: number;
    track: string;
    trackId: number;
    user: string;
    userId: string;
    count: number;
}

export interface MotecLapStat extends MotecStat {
    fastestLap: number;
    averageFastestLap: number;
    trackCondition: string;
    laptime: number;
}

export interface UserMetric extends MotecStat {
    favouriteCar: string;
    favouriteTrack: string;
    numberOfFastestLaps: number;
    numberOfLaps: number;
    numberOfUploads: number;
}